dhall/tests/Dhall/Test/TypeInference.hs

```diff
@@ -84,25 +84,32 @@ successTest prefix = do
         Tasty.HUnit.assertEqual message resolvedExpectedType inferredType
 
 failureTest :: Text -> TestTree
-failureTest path = do
-    let skip = [ typeInferenceDirectory </> "failure/unit/MergeEmptyNeedsDirectAnnotation1.dhall"
+failureTest prefix = do
+    let skip = [ typeInferenceDirectory </> "failure/unit/MergeEmptyNeedsDirectAnnotation1"
                ]
 
-    Test.Util.testCase path skip $ do
-        let dhallPath = Test.Util.toDhallPath path
+    Test.Util.testCase prefix skip $ do
+        let prefixFP = Text.unpack prefix
 
-        expression <- Core.throws (Parser.exprFromText mempty dhallPath)
+        code <- Text.IO.readFile (prefixFP <> ".dhall")
 
-        let io :: IO Bool
-            io = do
-                _ <- Test.Util.load expression
-                return True
+        case Parser.exprFromText mempty code of
+            Left _ -> return ()
+                      -- Duplicate fields are incorrectly caught during parsing:
```

Comment:
Instead of ignoring these parse errors, should we just skip the tests that fail in this way or mark them as expected failures?

Reply:
I've marked them as expected failures.